### ISSUE TITLE
Why does one subset call result in 8 openDataset calls?

### ISSUE THREAD

**Issue by rsignell-usgs** [nctoolbox/nctoolbox#57]
I was trying to figure out why subsetting is so slow -- the routine below, which does a geographic subset from one layer for one specific time step from an ocean model takes 50 seconds:
```
url='http://geoport.whoi.edu/thredds/dodsC/coawst_4/use/fmrc/coawst_4_use_best.ncd'
nc=ncgeodataset(url);
gvar=nc.geovariable('temp');
% pull a subsetted geographic region at a specific time
s.lon=[-71 -65];
s.lat=[39 44];
s.time=now-5;
s.z_index=1;
sub=gvar.geosubset(s)
pcolorjw(sub.grid.lon,sub.grid.lat,double(squeeze(sub.data)));
```
When I turn on the profiler (`>> profile viewer'), I find that these take up the majority of the time:
* 7 calls to `.NetcdfDataset.openDataset`
* 2 calls to `GridDataset.open`
Is this really necessary?

![1-16-2015 2-37-19 pm](https://cloud.githubusercontent.com/assets/1872600/5783040/bd05dc18-9d8d-11e4-8acb-7b516bb23fd9.png)
![1-16-2015 2-38-02 pm](https://cloud.githubusercontent.com/assets/1872600/5783041/c06d706e-9d8d-11e4-8a32-ff547f499f3b.png)
![1-16-2015 2-39-03 pm](https://cloud.githubusercontent.com/assets/1872600/5783043/c4b15d98-9d8d-11e4-9fdd-96563ba29a7b.png)

**Comment by hohonuuli** [comment]
Yep, that's bad. Good catch. First ncgeovariable shouldn't be opening datasets at all. Second, GridDataset has a constructor that takes a NetcdfDataset instead of a string location; I assume that will not open a new dataset but share the underlying one. So the lines like:

```matlab
griddataset = ucar.nc2.dt.grid.GridDataset.open(obj.dataset.location);
```

Could be changed to:

```matlab
griddataset = ucar.nc2.dt.grid.GridDataset(obj.dataset);
```

Try that and see if it improves the performance.

**Comment by hohonuuli** [comment]
Whoops, never mind. It looks like that constructor in GridDataset is introduced in a later version of NetCDF-Java than the one nctoolbox is using.

That's too bad 'cause the latest version of Netcdf aren't playing well with matlab: #52

**Comment by hohonuuli** [comment]
An alternative would be to keep a reference to the GridDataset in ncgeodataset. I'll try to look into this next week.

**Comment by acrosby** [comment]
If i remember correctly that is a work around for the way that netcdf-java treats some datasets that don't get opened as grid datasets. It may not be a very *good* work around...
